net_arp.c

```diff
@@ -0,0 +1,337 @@
+		rta = RTA_NEXT(rta, len);
+	}
+	return 0;
+}
+
+static int get_arp_len(int net_sock)
+{
+	int recvl, err;
+	struct msghdr msg;
+	struct iovec iov;
+	struct sockaddr_nl nladdr;
+
+	msg.msg_name = &nladdr;
+	msg.msg_namelen = sizeof(nladdr);
+	iov.iov_base = NULL;
+	iov.iov_len = 0;
+
+	msg.msg_iov = &iov;
+	msg.msg_iovlen = 1;
+
+	recvl = recvmsg(net_sock, &msg, MSG_PEEK | MSG_TRUNC);
+	if (recvl < 0) {
+		err = errno;
+		eprintf("Error in get length recvmsg(), errno %d\n", err);
+		errno = err;
+		return -1;
+	}
+#ifdef PRINTLOGS1
+	printf("received len %d\n",recvl);
+#endif
+	return recvl;
+}
+
+static int get_max_arp(int len)
+{
+	int header_size, max_arp;
+	header_size = (sizeof(struct nlmsghdr)) + (sizeof(struct ndmsg));
+	max_arp = len/header_size;
+	return max_arp;
+}
+
+#ifdef TESTING
+static int get_attr(struct rtattr *at[], struct arp_info *arp, struct ndmsg *n,
+		FILE *fp)
+#else
+static int get_attr(struct rtattr *at[], struct arp_info *arp, struct ndmsg *n)
+#endif
+{
+#ifdef TESTING
+	char ether_type[] = "ether";
+	char ifname[IF_NAMESIZE];
+#endif
+	bzero(arp, sizeof(*arp));
+	if (n->ndm_family == AF_INET) {
+		if (at[NDA_DST]) {
+			arp->arp_ip_addr_len = RTA_PAYLOAD(at[NDA_DST]);
+			memcpy(arp->arp_ip_addr, RTA_DATA(at[NDA_DST]), arp->arp_ip_addr_len);
+			arp->arp_ip_addr[arp->arp_ip_addr_len] = '\0';
+#ifdef PRINTLOGS
+			printf("ARP IP len: %lu\n",arp->arp_ip_addr_len);
+			printf("ARP IP ADDR: %hhu.%hhu.%hhu.%hhu\n", arp->arp_ip_addr[0], arp->arp_ip_addr[1], arp->arp_ip_addr[2], arp->arp_ip_addr[3]);
+#endif
+#ifdef TESTING
+			fprintf(fp, "(%hhu.%hhu.%hhu.%hhu) at ", arp->arp_ip_addr[0], arp->arp_ip_addr[1], arp->arp_ip_addr[2], arp->arp_ip_addr[3]);
+#endif
+		}
+		if (at[NDA_LLADDR]) {
+			arp->arp_phys_addr_len = RTA_PAYLOAD(at[NDA_LLADDR]);
+			memcpy(arp->arp_phys_addr, RTA_DATA(at[NDA_LLADDR]), arp->arp_phys_addr_len);
+#ifdef PRINTLOGS
+			printf("ARP PHYS len: %lu\n",arp->arp_phys_addr_len);
+			printf("ARP PHYS ADDR: %02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx\n", arp->arp_phys_addr[0], arp->arp_phys_addr[1], arp->arp_phys_addr[2], arp->arp_phys_addr[3], arp->arp_phys_addr[4], arp->arp_phys_addr[5]);
+#endif
+#ifdef TESTING
+			fprintf(fp, "%02hhx:%02x:%02hhx:%02hhx:%02hhx:%02hhx ", arp->arp_phys_addr[0], arp->arp_phys_addr[1], arp->arp_phys_addr[2], arp->arp_phys_addr[3], arp->arp_phys_addr[4], arp->arp_phys_addr[5]);
+#endif
+		}
+#ifdef TESTING
+		fprintf(fp, "[%s] on %s\n", ether_type, if_indextoname(n->ndm_ifindex, ifname));
+#endif
+	}
+	return 0;
+}
+
+static int handle_arp_resp(int net_sock, void **out)
+{
+	int err, recvl, len, max_arp, arp_size, ret = 0;
+	int aind = 0, total_arp_size = 0;
+	struct iovec iov;
+	struct msghdr msg;
+	struct nlmsghdr *r;
+	struct nlmsgerr *nlmsg_err;
+	struct ndmsg *nd_msg;
+	struct rtattr *at[NDA_MAX + 1], *rta;
+	struct arp_info *arps = NULL, *iarps = NULL;
+	struct sockaddr_nl nladdr;
+#ifdef TESTING
+	FILE *fp = NULL;
+	fp = fopen ("./arp_info.txt", "w");
```

Comment:
Now, this is just testing code, so probably no big deal, but you're not checking fopen return value

Reply:
If you're around and push a fix in the next 5m, great - else i'll just merge it as is for now.